c/planarityTestGraphFunctionality.c

```diff
@@ -72,10 +79,100 @@ int TestGraphFunctionality(char *commandString, char *infileName, char *inputStr
 			}
 	
 		}
-		// TODO: add elif for algorithm command handling
+		else if (strchr("pdo234", commandString[1]))
+		{
+			if (inputStr != NULL)
+			{
+				ErrorMessage("TestGraphFunctionality only supports applying chosen algorithm to graphs read from file at this time.\n");
+				Result = NOTOK;
+			}
+			else
+			{
+				if (infileName == NULL)
+				{
+					ErrorMessage("No input file provided.\n");
+					Result = NOTOK;
+				}
+				else
+				{
+					inputStr = ReadTextFileIntoString(infileName);
+
+					strOrFileP testOutput = NULL;
+
+					if (outfileName != NULL)
+					{
+						FILE *outputFileP = fopen(outfileName, "w");
+						if (outputFileP == NULL)
+						{
+							errorStr = "Unable to open file \"%s\" for output.\n";
+							sprintf(Line, errorStr, outfileName);
+							ErrorMessage(Line);
+							free(inputStr);
+							inputStr = NULL;
+							gp_Free(&theGraph);
+						}
+						
+						testOutput = sf_New(outputFileP, NULL);
+					}
+					else
+					{
+						if ((*outputStr) == NULL)
+							(*outputStr) = (char *) malloc(1 * sizeof(char));
+						
+						testOutput = sf_New(NULL, (*outputStr));
+					}
+
+					if (testOutput == NULL)
+					{
+						ErrorMessage("Unable to set up string-or-file container for test output.\n");
+						free(inputStr);
+						inputStr = NULL;
+						gp_Free(&theGraph);
+						sf_Free(&testOutput);
+						return NOTOK;
+					}
+
+					char *infileBasename = NULL;
+					char *finalSlash = strrchr(infileName, '/');
+					if (finalSlash == NULL)
+					{
+						finalSlash = strrchr(infileName, '\\');
+						if (finalSlash == NULL)
+						{
+							finalSlash = infileName;
+						}
+					}
+
+					infileBasename = finalSlash + 1;
```

Comment:
I think this should be written slightly differently for two reasons: 
1. You should be able to rely on FILE_DELIMITER in appconst.h to cut this down to one strrchr() call. Double-check that windows and mac are still working after this change. 
2. The current approach loses the first character when infileName contains no FILE_DELIMITER, so I'd then recommend getting rid of the whole if construct and just using a ternary conditional in the assignment to infileBasename. Specifically, `infileBasename = finalSlash ? finalSlash+1 : infileName;`.